src/haversine.jl

```diff
@@ -0,0 +1,42 @@
+"""
+    Haversine(radius)
+
+The haversine distance between two locations on a sphere of given `radius`.
+
+Locations are described with longitude and latitude in degrees and
+the radius of the Earth is used by default (≈ 6371km). The computed
+distance has the same units as that of the radius.
+
+### Notes
+
+The haversine formula is widely used to approximate the geodesic distance
+between two points at the surface of the Earth. The error from approximating
+the Earth as a sphere is typically negligible for most applications. It is
+no more than 0.3%.
+"""
+struct Haversine{T<:Real} <: Metric
+  radius::T
```

Comment:
Please use four spaces for indentation in keeping with the rest of the repo. In general, the stylistic conventions in JuliaStats tries to mirror those in Base.